py/dml/output.py

```diff
@@ -187,13 +187,27 @@ def coverity_markers(markers, site=None):
     if dml.globals.coverity and site_with_loc:
         custom_markers = []
         filename = site.filename()
-        lineno = site.lineno
+        tgt_lineno = site.lineno
 
-        while (filename, lineno) in dml.globals.coverity_pragmas:
-            (lineno,
-             inline_markers) = dml.globals.coverity_pragmas[(filename, lineno)]
+        while (filename, tgt_lineno) in dml.globals.coverity_pragmas:
+            (start_lineno,
+             inline_markers) = dml.globals.coverity_pragmas[(filename,
+                                                             tgt_lineno)]
             custom_markers.extend(reversed(inline_markers))
 
+            # A minor HACK to handle the case of:
+            #   /*% COVERITY foo %*/ /*% COVERITY
+            #   bar %*/
+            #   some_statement
```

Comment:
what if you add more weirdness, like
```
/*% COVERITY foo %*/ /*%

COVERITY
bar %*/ /*

COVERITY

baz
%*/
statement
```
? do you have all corners covered?

Reply:
Not to mention:
```
/*% COVERITY foo %*/ /*%
foo %*/
statement
```

Reply:
The only corner you mention not covered is
```
/*% COVERITY foo %*/ /*%
foo %*/
statement
```
which results in `statement` receiving no annotation, as `foo` is not a `COVERITY` pragma. This might change once we change and generalize the line-association mechanism to work with pragmas in general. However, that'd only be true for **pragmas**; not comments, so in the case of: 
```
/*% COVERITY foo %*/ /*
foo */
statement
```
then `statement` would not receive an annotation, not today and not ever. At least, not in my vision of how pragmas would work.